Sherlock/yoda/DOCU.md

```diff
@@ -381,48 +401,54 @@ api.Transaction([](PrimesAPI::T_DATA data) {
   const auto accessor = Matrix<PrimeCell>::Accessor(data);
   EXPECT_EQ(10u, accessor.Rows().size());
   EXPECT_EQ(6u, accessor.Cols().size());  // 1, 2, 3, 5, 7 or 9, my captain.
-  std::string by_rows_keys;
-  std::string by_rows_values;
+  std::vector<std::string> by_rows_keys;
+  std::vector<std::string> by_rows_values;
   for (const auto cit : accessor.Rows()) {
-    by_rows_keys += Printf("[`%d`:%d]",
+    by_rows_keys.push_back(Printf("[`%d`:%d]",
                            static_cast<int>(cit.key()),
-                           static_cast<int>(cit.size()));
-    by_rows_values += "[";
+                           static_cast<int>(cit.size())));
+    std::string v = "[";
     bool first = true;
     for (const auto cit2 : cit) {
-      by_rows_values += Printf("%s%d", first ? "" : ",", cit2.index);
+      v += Printf("%s%d", first ? "" : ",", cit2.index);
       first = false;
     }
-    by_rows_values += "]";
+    v += "]";
+    by_rows_values.push_back(v);
   }
+  std::sort(begin(by_rows_keys), end(by_rows_keys));
```

Comment:
OCD goes both ways!

You could:
```
std::set<std::string> by_rows_keys_set(by_row_keys.begin(), by_row_keys().end());
rows_unordeded |= (Join(by_rows_keys) != Join(by_rows_keys_set));
```
and test `by_rows_keys_set` immediately below and `bool rows_unordeded` after all the rows are scanned. And same for cols.